### PR TITLE
feat(transaction): simplify calc multi trans unique shard count

### PR DESCRIPTION
**Pull request by adiholden**

### CODE REVIEW

File: src/server/transaction.cc
Comment (romange) [inline src/server/transaction.cc:652]: `auto&` -> `bool` pls

File: src/server/transaction.cc
Comment (romange) [inline src/server/transaction.cc:653]: `true == was_shard_write` -> `was_shard_write`

File: src/server/transaction.h
Comment (romange) [inline src/server/transaction.h:273]: explain what's the meaning of this field.

File: src/server/transaction.cc
Comment (romange) [inline src/server/transaction.cc:1099]: comparing to `true` - is it by purpose?

File: src/server/transaction.cc
Comment (romange) [inline src/server/transaction.cc:651]: can you explain what's the semantics of this function and why override `unique_shard_cnt_` for that?
i.e. how this is different from a "regular" `unique_shard_cnt_` ?

Also, even we need it, why do you need to store it inside unique_shard_cnt_ and not just return from this function and use this function when you need to know the value?

Comment (romange) [reply src/server/transaction.cc:651]: Overriding the same variable with different semantics makes me nervous. This is why I am asking these questions.

File: src/server/transaction.h
Comment (dranikpg) [reply src/server/transaction.h:273]: We should probably use an atomic bitset in the future

File: src/server/transaction.cc
Comment (adiholden) [reply src/server/transaction.cc]: I updated the comments in the header 
It made sense to me to update the unique_shard_cnt_ var to hold the shard count in the whole transaction but as it is only updated in case of journal write I updated in the new commit to have this value returned by the function

**Comment by adiholden** [comment]
@romange waiting for your approval

**Review by romange — APPROVED** [review]

**Review by romange — APPROVED** [review]

**Pull request merged**
